tools/source_updater/updater/main.py

```diff
@@ -0,0 +1,80 @@
+import glob
+import logging
+import os
+
+from airbyte_cdk.connector import BaseConnector
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
+from updater.catalog import CatalogAssembler, CatalogMerger
+from updater.config import Config
+from updater.handler import SourceUpdaterHandler
+from updater.source import SourceRepository
+
+_DEFAULT_CONFIG_FILE = "config.json"
+
+logger = logging.getLogger("main")
+
+
+def _assemble_configs(config_path: str):
+    """
+    :return: the main config and the set of the other configs
+    """
+    if os.path.isfile(config_path):
+        return Config(
+            "config",
+            BaseConnector.read_config(
+                config_path if config_path.endswith(_DEFAULT_CONFIG_FILE) else os.path.join(config_path, _DEFAULT_CONFIG_FILE)
+            )
+        ), set()
+
+    config_files = glob.glob(os.path.join(config_path, "*.json"))
+    main_config = None
+    other_configs = set()
+    for file in config_files:
+        if os.path.basename(file) == _DEFAULT_CONFIG_FILE:
+            main_config = Config(
+                "config",
+                BaseConnector.read_config(
+                    config_path if config_path.endswith(_DEFAULT_CONFIG_FILE) else os.path.join(config_path, _DEFAULT_CONFIG_FILE)
+                )
+            )
+        else:
+            other_configs.add(
+                Config(
+                    os.path.basename(file).replace(".json", ""),
+                    BaseConnector.read_config(
+                        config_path if config_path.endswith(_DEFAULT_CONFIG_FILE) else os.path.join(config_path, _DEFAULT_CONFIG_FILE)
+                    )
+                )
+            )
+
+    if not main_config:
+        raise ValueError(f"Could not find main config. Name should match {_DEFAULT_CONFIG_FILE}")
+    return main_config, other_configs
+
+
+if __name__ == "__main__":
+    import argparse
```

Comment:
very nit: I recommend using click for new scripts. I find it much nice than argparse https://github.com/airbytehq/airbyte/blob/1c871756b31be55fc417e0df8f87f6ae9c842074/tools/ci_credentials/ci_credentials/main.py#L19-L22